mayday.go

```diff
@@ -2,289 +2,94 @@ package main
-			Args: []string{"systemd-cgls"},
-		},
-		{
-			Args: []string{"systemd-cgtop", "-n1"},
-		},
-		{
-			Args: []string{"ps", "fauxwww"},
-			Link: "ps",
-		},
-		{
-			Args: []string{"lsmod"},
-			Link: "lsmod",
-		},
-		{
-			Args: []string{"lspci"},
-			Link: "lspci",
-		},
-		{
-			Args: []string{"lsof", "-b", "-M", "-n", "-l"},
-			Link: "lsof",
-		},
-		{
-			Args: []string{"blkid"},
-		},
-		{
-			Args: []string{"btrfs", "fi", "show"},
-		},
-		{
-			Args: []string{"df", "-al"},
-			Link: "df",
-		},
-		{
-			Args: []string{"ip", "-o", "addr", "show"},
-			Link: "ip_addr_show",
-		},
-		{
-			Args: []string{"ip", "-o", "link", "show"},
-			Link: "ip_link_show",
-		},
-		{
-			Args: []string{"ip", "-o", "route", "show"},
-			Link: "ip_route_show",
-		},
-		{
-			Args: []string{"netstat", "-neopa"},
-			Link: "netstat",
-		},
-		{
-			Args: []string{"df", "-ali"},
-		},
-		{
-			Args: []string{"free", "-m"},
-			Link: "free",
-		},
-		{
-			Args: []string{"systemctl", "list-units", "-a"},
-			Link: "all_units",
-		},
-		{
-			Args: []string{"systemctl", "list-units", "--state=running"},
-			Link: "running_units",
-		},
-		{
-			Args: []string{"systemctl", "status", "etcd.service"},
-			Link: "etcd_status",
-		},
-		{
-			Args: []string{"systemctl", "status", "etcd2.service"},
-			Link: "etcd2_status",
-		},
-		{
-			Args: []string{"systemctl", "status", "fleet.service"},
-			Link: "fleet_status",
-		},
-		{
-			Args: []string{"systemctl", "status", "flanneld.service"},
-			Link: "flanneld_status",
-		},
-	}
-)
+type Config struct {
+	Files    []mayday.File
+	Commands []mayday.Command
+}
 
-func getJournals(dir string) error {
-	c, err := dbus.New()
+func readConfig() ([]mayday.File, []mayday.Command, error) {
+	dir, err := filepath.Abs(filepath.Dir(os.Args[0]))
 	if err != nil {
-		return err
+		log.Fatal(err)
 	}
+	log.Printf(dir)
+	configFile := dir + "/../config.json"
 
-	defer c.Close()
-
-	units, err := c.ListUnits()
+	log.Printf("Reading configuration from %v\n", configFile)
+	dat, err := ioutil.ReadFile(configFile)
 	if err != nil {
-		return err
+		log.Fatal(err)
 	}
 
-	pathre := regexp.MustCompile(`/usr/lib(32|64)?/systemd/system/.*\.service`)
-
-	var svcs []string
+	var c Config
 
-	// build a list of units that live in /usr/lib/system/system
-	for _, u := range units {
-		if p, err := c.GetUnitProperty(u.Name, "FragmentPath"); err == nil {
-			path := p.Value.Value().(string)
-
-			if pathre.MatchString(path) {
-				svcs = append(svcs, u.Name)
-			}
-		}
+	err = json.Unmarshal(dat, &c)
+	if err != nil {
+		log.Fatal(err)
 	}
+	return c.Files, c.Commands, nil
+}
 
-	// get the journals of the units
-	for _, svc := range svcs {
-		daysago := 7
-
-		logfile := path.Join(dir, fmt.Sprintf("%s.log", svc))
+func main() {
 
-		log.Printf("collecting %d days of logs from %q", daysago, svc)
+	files, commands, err := readConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
 
-		out, err := os.Create(logfile)
+	for i := 0; i < len(files); i++ {
```

Comment:
I think it'd make more sense to the config contain filepaths, rather than mayday.Files, and have this loop construct mayday.Files, rather than fill out incomplete mayday.Files returned from config.  In general I think it's best to try to limit access to incomplete states as much as possible, ideally to just within constructor functions.